builtin/credential/ldap/path_config.go

```diff
@@ -367,131 +358,6 @@ func (b *backend) pathConfigWrite(ctx context.Context, req *logical.Request, d *
 	return nil, nil
 }
 
-type ConfigEntry struct {
-	logger        log.Logger
-	Url           string `json:"url"`
-	UserDN        string `json:"userdn"`
-	GroupDN       string `json:"groupdn"`
-	GroupFilter   string `json:"groupfilter"`
-	GroupAttr     string `json:"groupattr"`
-	UPNDomain     string `json:"upndomain"`
-	UserAttr      string `json:"userattr"`
-	Certificate   string `json:"certificate"`
-	InsecureTLS   bool   `json:"insecure_tls"`
-	StartTLS      bool   `json:"starttls"`
-	BindDN        string `json:"binddn"`
-	BindPassword  string `json:"bindpass"`
-	DenyNullBind  bool   `json:"deny_null_bind"`
-	DiscoverDN    bool   `json:"discoverdn"`
-	TLSMinVersion string `json:"tls_min_version"`
-	TLSMaxVersion string `json:"tls_max_version"`
-
-	// This json tag deviates from snake case because there was a past issue
-	// where the tag was being ignored, causing it to be jsonified as "CaseSensitiveNames".
-	// To continue reading in users' previously stored values,
-	// we chose to carry that forward.
-	CaseSensitiveNames *bool `json:"CaseSensitiveNames,omitempty"`
-}
-
-func (c *ConfigEntry) GetTLSConfig(host string) (*tls.Config, error) {
-	tlsConfig := &tls.Config{
-		ServerName: host,
-	}
-
-	if c.TLSMinVersion != "" {
-		tlsMinVersion, ok := tlsutil.TLSLookup[c.TLSMinVersion]
-		if !ok {
-			return nil, fmt.Errorf("invalid 'tls_min_version' in config")
-		}
-		tlsConfig.MinVersion = tlsMinVersion
-	}
-
-	if c.TLSMaxVersion != "" {
-		tlsMaxVersion, ok := tlsutil.TLSLookup[c.TLSMaxVersion]
-		if !ok {
-			return nil, fmt.Errorf("invalid 'tls_max_version' in config")
-		}
-		tlsConfig.MaxVersion = tlsMaxVersion
-	}
-
-	if c.InsecureTLS {
-		tlsConfig.InsecureSkipVerify = true
-	}
-	if c.Certificate != "" {
-		caPool := x509.NewCertPool()
-		ok := caPool.AppendCertsFromPEM([]byte(c.Certificate))
-		if !ok {
-			return nil, fmt.Errorf("could not append CA certificate")
-		}
-		tlsConfig.RootCAs = caPool
-	}
-	return tlsConfig, nil
-}
-
-func (c *ConfigEntry) DialLDAP() (*ldap.Conn, error) {
```

Comment:
This was moved to the client with no change except that the logger is passed in rather than found on the config.